### DISCUSSION TITLE
Rate

### DISCUSSION THREAD

**Discussion by wokkeltje13** [jomjol/AI-on-the-edge-device#810] [Q&A]
What does the value of "rate" means?

In MQTT I receive the following: 
`{"value":4499.9,"raw":"004499.9","error":"no error","rate":0.000000,"timestamp":"2022-05-20T21:03:02"}`

So a rate of 0.026693 when the previous value was 4499.8 and the new was 4499.9 so I would expect a rate of 0.1 but not something with 6 decimals.

The reason for my question is that I am looking for the difference between 2 reading + and - to incorporate in my HA energy dashboard (I need home to grid and grid to home)

Where do I make a mistake with the meaning of rate?

Thanks

**Comment by friedpa** [comment]
Maybe your MQTT sample is wrong, the rate above is 0 which is correct, in your sample, which you described afterwards in the text, the time between the two measurements are missing to calculate the rate and see if there is an error. The rate is related to a consumption (e.g. water) per timeframe (between the measurement and the previous measurement.

**Reply by wokkeltje13** [reply]
indeed, the example was wrong, but I think I found the solution myselve.

the rate is calculated in minutes I think, so 0.1/4.85

Will "AbsoluteChange" return the effective change between 2 measurements?

**Comment by friedpa** [comment]
I always use `Rate`, but I think `AbsoluteChange` is the Value between two measurements, see Wiki for details.

**Comment by wokkeltje13** [comment]
AbsoluteChange also gives me a calculated value. I should need the value between 2 measurements without any calculation. (But this is no rate them off course)

**Comment by jomjol** [comment]
The rate is a real rate = change/time. Here time = 1 minute.
If you want to calculate the absolute change to can mulitply the rate with the time difference from the last to the current time stamp in minutes.

**Comment by wokkeltje13** [comment]
The issue is that in HA, you can't get the previous value (easily). If that was possible I could easily count the delta between the new and old value without the need of multiplying the rate with the time difference.
So it would be nice to have the delta value also be transfered to MQTT.

**Comment by danube** [comment]
Haven't found any description about rate except this one here. It may be a nice idea to add this to the manual. I'd also suggest to add `main/changeabsolut` and `interval` to the json object. #2335

**Reply by danube** [reply]
Suggested change of MQTT API documentation: https://github.com/jomjol/AI-on-the-edge-device-docs/pull/33
